src/devices/cddasongloader.cpp

```diff
@@ -30,10 +30,19 @@ CddaSongLoader::CddaSongLoader(const QUrl& url, QObject* parent)
     : QObject(parent), url_(url), cdda_(nullptr), cdio_(nullptr) {
   connect(this, SIGNAL(MusicBrainzDiscIdLoaded(const QString&)),
           SLOT(LoadAudioCDTags(const QString&)));
+
+  cdio_ = cdio_open(url_.path().toLocal8Bit().constData(), DRIVER_DEVICE);
+  Q_ASSERT(cdio_ != nullptr);  // todo: assert for now, but we need to actually
+                               // handle error cases here
```

Comment:
Maybe we can add a separate Init() method that returns a bool? CddaDevice already has an Init method that could call it, though it doesn't currently return anything.

Reply:
I'm not a fan of Init() methods if they can be avoided. Having an instance of CddaSongLoader that is not in a properly initialised state (i.e. where `Init()` would have returned `false`) seems pointless because it's unusable. Better to fail during construction, imo.